Cannot compile datatable with gcc

- How to reproduce the bug?
```
git clone https://github.com/h2oai/datatable.git
cd datatable
unset LLVM{4,5,6,7} 
make
```

fails on
```
c/extras/py_ftrl.cc:220:32: error: the value of 'py::fn___getstate__' is not usable in a constant expression
   mm.add<&Ftrl::m__getstate__, fn___getstate__>();
                                ^~~~~~~~~~~~~~~
c/extras/py_ftrl.cc:39:15: note: 'py::fn___getstate__' was not declared 'constexpr'
 static NoArgs fn___getstate__("__getstate__", nullptr);
               ^~~~~~~~~~~~~~~
c/extras/py_ftrl.cc:221:32: error: the value of 'py::fn___setstate__' is not usable in a constant expression
   mm.add<&Ftrl::m__setstate__, fn___setstate__>();
                                ^~~~~~~~~~~~~~~
c/extras/py_ftrl.cc:40:15: note: 'py::fn___setstate__' was not declared 'constexpr'
 static PKArgs fn___setstate__(1, 0, 0, false, false, {"state"},
               ^~~~~~~~~~~~~~~
```

- What was the expected behavior?
The same as in 0.7.0 - i would get binary to compile

- Your environment?
Ubuntu python3.6.7, virtualenv

```
(test) >:~$ python --version
Python 3.6.7

(test) >:~$ pip list
Package         Version Location
--------------- ------- -------------------
blessed         1.15.0
colorama        0.4.1
cycler          0.10.0
kiwisolver      1.0.1
matplotlib      3.0.2   /tmp/matplotlib/lib
numpy           1.16.0
pip             19.0.1
pkg-resources   0.0.0
pyparsing       2.3.1
python-dateutil 2.7.5
setuptools      40.6.3
six             1.12.0
typesentry      0.2.7
wcwidth         0.1.7
```

@st-pasha are we going to support gcc? If so, it could be a good idea to add the profile described by @mmalohlava to our normal check list. As, for example, even when I disable the failing part of FTRL code, it fails with

```
c/frame/py_frame.cc: In static member function ‘static void py::Frame::Type::init_methods_and_getsets(py::ExtType<py::Frame>::Methods&, py::ExtType<py::Frame>::GetSetters&)’:
c/frame/py_frame.cc:131:24: error: the value of ‘py::fn_head’ is not usable in a constant expression
   mm.add<&Frame::head, fn_head>();
                        ^~~~~~~
c/frame/py_frame.cc:49:15: note: ‘py::fn_head’ was not declared ‘constexpr’
 static PKArgs fn_head(
               ^~~~~~~
c/frame/py_frame.cc:132:24: error: the value of ‘py::fn_tail’ is not usable in a constant expression
   mm.add<&Frame::tail, fn_tail>();
                        ^~~~~~~
c/frame/py_frame.cc:58:15: note: ‘py::fn_tail’ was not declared ‘constexpr’
 static PKArgs fn_tail(
```

--- a/c/extras/py_ftrl.cc
+++ b/c/extras/py_ftrl.cc
@@ -36,9 +36,9 @@ PKArgs Ftrl::Type::args___init__(0, 2, 7, false, false,
                                  "lambda2", "nbins", "nepochs", "interactions"},
                                  "__init__", nullptr);
 
-static NoArgs fn___getstate__("__getstate__", nullptr);
-static PKArgs fn___setstate__(1, 0, 0, false, false, {"state"},
-                              "__setstate__", nullptr);
+NoArgs Ftrl::Type::fn___getstate__("__getstate__", nullptr);
+PKArgs Ftrl::Type::fn___setstate__(1, 0, 0, false, false, {"state"},
+                                   "__setstate__", nullptr);
 
 static const char* doc_alpha        = "`alpha` in per-coordinate FTRL-Proximal algorithm";
 static const char* doc_beta         = "`beta` in per-coordinate FTRL-Proximal algorithm";
@@ -162,7 +162,7 @@ void Ftrl::m__init__(PKArgs& args) {
 void Ftrl::init_dtft(dt::FtrlParams dt_params) {
   size_t nlabels = labels.size();
   xassert(nlabels > 0);
-  // If there is only two labels provided, we only need 
+  // If there is only two labels provided, we only need
   // one classifier.
   size_t ndtft = nlabels - (nlabels == 2);
   dtft->clear();
@@ -439,7 +439,7 @@ void Ftrl::fit_multinomial(DataTable* dt_X, DataTable* dt_y) {
   }
 
   // Train all the classifiers. NB: when there is two labels,
-  // `init_dtft()` creates only one classifier, just like in a 
+  // `init_dtft()` creates only one classifier, just like in a
   // binomial case.
   size_t ndtft = dtft->size();
   for (size_t i = 0; i < ndtft; ++i) {

--- a/c/extras/py_ftrl.h
+++ b/c/extras/py_ftrl.h
@@ -52,6 +52,8 @@ class Ftrl : public PyObject {
         static PKArgs args_fit;
         static PKArgs args_predict;
         static NoArgs args_reset;
+        static NoArgs fn___getstate__;
+        static PKArgs fn___setstate__;
         static const char* classname();
         static const char* classdoc();
         static bool is_subclassable() { return true; }

--- a/c/frame/__init__.cc
+++ b/c/frame/__init__.cc
@@ -685,9 +685,9 @@ void Frame::m__init__(PKArgs& args) {
 // pickling / unpickling
 //------------------------------------------------------------------------------
 
-static NoArgs fn___getstate__("__getstate__", nullptr);
-static PKArgs fn___setstate__(1, 0, 0, false, false, {"state"},
-                              "__setstate__", nullptr);
+NoArgs Frame::Type::fn___getstate__("__getstate__", nullptr);
+PKArgs Frame::Type::fn___setstate__(1, 0, 0, false, false, {"state"},
+                                    "__setstate__", nullptr);
 
 
 // TODO: add py::obytes object

--- a/c/frame/py_frame.cc
+++ b/c/frame/py_frame.cc
@@ -46,7 +46,7 @@ NoArgs Frame::Type::args_copy("copy",
 "Thus, for all intents and purposes the copied Frame will behave as if\n"
 "it was deep-copied.\n");
 
-static PKArgs fn_head(
+PKArgs Frame::Type::fn_head(
     1, 0, 0, false, false,
     {"n"}, "head",
 R"(head(self, n=10)
@@ -55,7 +55,7 @@ R"(head(self, n=10)
 Return the first `n` rows of the Frame, same as ``self[:n, :]``.
 )");
 
-static PKArgs fn_tail(
+PKArgs Frame::Type::fn_tail(
     1, 0, 0, false, false,
     {"n"}, "tail",
 R"(tail(self, n=10)

--- a/c/frame/py_frame.h
+++ b/c/frame/py_frame.h
@@ -61,6 +61,8 @@ class Frame : public PyObject {
         static NoArgs args_to_tuples;
         static PKArgs fn_head;
         static PKArgs fn_tail;
+        static NoArgs fn___getstate__;
+        static PKArgs fn___setstate__;
         static const char* classname();
         static const char* classdoc();
         static bool is_subclassable() { return true; }

--- a/c/frame/py_frame.h
+++ b/c/frame/py_frame.h
@@ -59,6 +59,8 @@ class Frame : public PyObject {
         static NoArgs args_to_dict;
         static NoArgs args_to_list;
         static NoArgs args_to_tuples;
+        static PKArgs fn_head;
+        static PKArgs fn_tail;
         static const char* classname();
         static const char* classdoc();
         static bool is_subclassable() { return true; }

